server/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java

```diff
@@ -490,6 +490,16 @@ private static void checkEnabledFieldChange(ObjectMapper mergeWith, Mapper merge
                 final String path = mergeWith.fullPath() + "." + mergeWithObjectMapper.simpleName() + ".enabled";
                 throw new MapperException("Can't update attribute for type [" + path + "] in index mapping");
             }
+
+            if (mergeIntoObjectMapper.nested().isIncludeInParent() != mergeWithObjectMapper.nested().isIncludeInParent()) {
+                final String path = mergeWith.fullPath() + "." + mergeWithObjectMapper.simpleName() + ".include_in_parent";
+                throw new MapperException("Can't update attribute for type [" + path + "] in index mapping");
```

Comment:
I think a message like this would be clearer: "Merging mappings failed for the [user] field: the [include_in_parent] parameter cannot be updated."